src/components/screens/map-screen/map-screen.tsx

```diff
@@ -1048,7 +1054,8 @@ class MapScreen extends React.Component<Props, State> {
         newCenter[0] = newCenterCoordinates[0];
         newCenter[1] = newCenterCoordinates[1];
       }
-
+      
+      this.updateRouteTotalAirQualityData(route);
```

Comment:
```suggestion
      await this.updateRouteTotalAirQualityData(route);
```
Await so it waits for the data before showing the route